### ISSUE TITLE
Add a signature box to the signed pdf

### ISSUE THREAD

**Issue by Chunkygoo** [vbuch/node-signpdf#226]
Is there a way to add a "signature box" to the signed pdf where if clicked on, shows the information of the certificate? 

Example:
![image](https://github.com/vbuch/node-signpdf/assets/52672536/5f69ce1d-3df3-4852-865b-db15713382a0)

Link:

https://ironpdf.com/blog/compare-to-other-components/add-digital-signature-topdf-in-csharp-using-itextsharp/

**Comment by vbuch** [comment]
Hi,
1: If the signature is valid or not is not supposed to be rendered in the document. That is for the reader (Adober Reader or other) to verify and tell you.
2: Adding a box is something standard. Here is [an example of how that's done with PDFKit](https://github.com/vbuch/node-signpdf/blob/develop/packages/examples/src/pdfkit010-with-visual.js). Also have a look at [the issues tagged "visual"](https://github.com/vbuch/node-signpdf/issues?q=label%3Avisual).

**Comment by Chunkygoo** [comment]
> Hi,
> 
> 1: If the signature is valid or not is not supposed to be rendered in the document. That is for the reader (Adober Reader or other) to verify and tell you.
> 
> 2: Adding a box is something standard. Here is [an example of how that's done with PDFKit](https://github.com/vbuch/node-signpdf/blob/develop/packages/examples/src/pdfkit010-with-visual.js). Also have a look at [the issues tagged "visual"](https://github.com/vbuch/node-signpdf/issues?q=label%3Avisual).

Thanks for the reply. I agree with you on 1, but the link I attached seems to be able to link the box to Adobe reader's "signature panel". Any idea how that's done?

**Comment by Chunkygoo** [comment]
which widget exactly? is it the addVisual function you're talking about? https://github.com/vbuch/node-signpdf/blob/develop/packages/examples/src/pdfkit010-with-visual.js#L82


Also, do you mind showing the full dialog of the 'Signature Validation Status' image please just so I can make sure it is indeed what Im looking for?

**Comment by vbuch** [comment]
No. I'm not talking about `addVisual`. That just draws something. I used the word "widget" intentfully. I don't think it can be more readable really: https://github.com/vbuch/node-signpdf/blob/develop/packages/examples/src/pdfkit010-with-visual.js#L96

[README here](https://github.com/vbuch/node-signpdf/blob/develop/packages/examples/README.md) then `node src/pdfkit010-with-visual.js`. Output will be in the `output` dir. Then you can make sure it is what you are looking for.

**Comment by troyam** [comment]
What version of node are you using?

**Comment by lutfi-haslab** [comment]
How to add image or annotation with pdf-lib? As far as i know, pdfkit only for creation, cause i dont find load function like in pdf-lib.

**Comment by vbuch** [comment]
@troyam 
```% node --version
v21.1.0```

@lutfi-haslab pdf-lib allows you to load. It also allows you to add whatever you want to the document. The only thing missing is [incremental updates](https://github.com/Hopding/pdf-lib/discussions/998) so multiple signatures are not possible with it. If you are asking about pdfkit (you wrote pdf-lib twice), I'm sorry but I don't know a way to do it with pdfkit. That is actually why the plain placeholder helper exists.
